### PR TITLE
OSDOCS-3030: Added AWS Marketplace install content

### PR DESCRIPTION
**Pull request by mjpytlak**
Version(s):
CP to 4.10, 4.11 and 4.12

Issue:
This PR addresses [osdocs-3030](https://issues.redhat.com/browse/OSDOCS-3030).

Link to docs preview:

- [Supported AWS Marketplace regions](http://file.rdu.redhat.com/mpytlak/osdocs-3030/installing/installing_aws/installing-aws-account.html#installation-aws-marketplace_installing-aws-account)
- Installing a cluster on AWS with customization > [Obtaining an AWS Marketplace image](http://file.rdu.redhat.com/mpytlak/osdocs-3030/installing/installing_aws/installing-aws-customizations.html#installation-aws-marketplace-subscribe_installing-aws-customizations)
- Installing a cluster on AWS into a government region > [Obtaining an AWS Marketplace image](http://file.rdu.redhat.com/mpytlak/osdocs-3030/installing/installing_aws/installing-aws-government-region.html#installation-aws-marketplace-subscribe_installing-aws-government-region)
- Installing a cluster on AWS using CloudFormation templates > [Obtaining an AWS Marketplace image](http://file.rdu.redhat.com/mpytlak/osdocs-3030/installing/installing_aws/installing-aws-user-infra.html#installation-aws-marketplace-subscribe_installing-aws-user-infra)
- Installing a cluster on AWS using CloudFormation templates > [Creating the worker nodes in AWS](http://file.rdu.redhat.com/mpytlak/osdocs-3030/installing/installing_aws/installing-aws-user-infra.html#installation-creating-aws-worker_installing-aws-user-infra)

### CODE REVIEW

**Comment by mjpytlak** [comment]
@jeana-redhat Wanted to give you a look at this ahead of our call next week.

**Comment by mjpytlak** [comment]
@patrickdillon @yunjiang29 Ready for SME and QE review. PTAL.

CC: @dmesser @jeana-redhat

File: modules/installation-aws-marketplace-subscribe.adoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+compute:
+- hyperthreading: Enabled
+    name: worker
+    platform:
+      aws:
+        amiID: ami-06c4d345f7c207239 <1>
+        type: m5.4xlarge
+    replicas: 3
+  metadata:
+    name: test-cluster
```
Comment (yunjiang29) [inline modules/installation-aws-marketplace-subscribe.adoc:48]: Is it something wrong with the formatter? It should be 

```
compute:
- hyperthreading: Enabled
  name: worker
  platform:
    aws:
      amiID: ami-06c4d345f7c207239 <1>
      type: m5.4xlarge
  replicas: 3
metadata:
  name: test-cluster
```

File: modules/installation-aws-marketplace-subscribe.adoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_aws/installing-aws-customizations.adoc
+// * installing/installing_aws/installing-aws-user-infra.adoc
```
Comment (yunjiang29) [inline modules/installation-aws-marketplace-subscribe.adoc:4]: Looks like it's missing in GovCloud section: http://file.rdu.redhat.com/mpytlak/osdocs-3030/installing/installing_aws/installing-aws-government-region.html

**Review by patrickdillon — COMMENTED** [review]
LGTM but one small nit/suggestion on wording

File: modules/installation-aws-marketplace-subscribe.adoc (hunk 1 above)
Comment (mjpytlak) [reply modules/installation-aws-marketplace-subscribe.adoc:48]: Bad copy/paste on my part. Fixed.

File: modules/installation-aws-marketplace-subscribe.adoc (hunk 2 above)
Comment (mjpytlak) [reply modules/installation-aws-marketplace-subscribe.adoc:4]: Thank you. Nice catch. Added to GovCloud topic.

**Comment by mjpytlak** [comment]
Thank you for the review @patrickdillon and @yunjiang29. Incorporated your feedback. Please take one more look.

**Comment by mjpytlak** [comment]
@dmesser This PR will merge to the 4.10 and 4.11 GA doc as soon as it is approved by @patrickdillon and @yunjiang29 and a doc peer review is complete. A separate effort is required to back port this work into 4.8 and 4.9.

@bscott-rh While the steps to install with an AWS remain the same for 4.8 and 4.9, the doc is structured slightly differently in these releases. Specifically - there is a single help topic for Gov/Secret regions in 4.8 and 4.9, while beginning with 4.10, Gov has its own assembly. You will need to back port these changes to 4.8 and 4.9 individually.

**Comment by yunjiang29** [comment]
@mjpytlak thanks for the updates. /lgtm

**Comment by patrickdillon** [comment]
LGTM

File: modules/installation-aws-marketplace-subscribe.adoc [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-east-2 <2>
+sshKey: ssh-ed25519 AAAA...
+pullSecret: '{"auths": ...}'
+----
+<1> The AMI id from your AWS Marketplace subscription.
+<2> Your AMI id is associated with a specific AWS region. When creating the installation configuration file, be sure to select the same AWS region that you specified when configuring your entitlement.
```
Comment (abhatt-rh) [inline modules/installation-aws-marketplace-subscribe.adoc:60]: https://docs.engineering.redhat.com/display/CCSENABLE/Phrasal+verbs
```suggestion
<2> Your AMI id is associated with a specific AWS region. When creating the installation configuration file, ensure that you select the same AWS region that you specified when configuring your entitlement.
```

**Review by abhatt-rh — COMMENTED** [review]
Hi Mike, mostly looks good! I added some comments for your consideration. Thanks

File: modules/installation-aws-marketplace-subscribe.adoc [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+* You have an AWS account to purchase the offer. This account does not have to be the same account that is used to install the cluster.
+
+.Procedure
+
+. Complete the {product-title} subscription from the link:https://aws.amazon.com/marketplace/fulfillment?productId=59ead7de-2540-4653-a8b0-fa7926d5c845[AWS Marketplace].
+ifdef::ipi[]
+. Record the AMI id for your specific region. As part of the installation process, you must update the `install-config.yaml` file with this value before deploying the cluster.
+endif::ipi[]
+ifdef::upi[]
+. Record the AMI id for your specific region. If you use the CloudFormation template to deploy your worker nodes, you can update the `worker0.type.properties.ImageID` parameter with this value.
```
Comment (jeana-redhat) [inline modules/installation-aws-marketplace-subscribe.adoc:33]: Just to verify, the other flavor of this is "you must" but this is "you can" - want to make sure that's as intended :)

**Review by jeana-redhat — COMMENTED** [review]
Just one additional thing I wondered about. Avani covered everything else :)

File: modules/installation-aws-marketplace-subscribe.adoc [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+:ipi:
+endif::[]
+ifeval::["{context}" == "installing-aws-user-infra"]
+:upi:
+endif::[]
+
+:_content-type: PROCEDURE
+[id="installation-aws-marketplace-subscribe_{context}"]
+= Obtaining an AWS Marketplace image
+If you are deploying an {product-title} cluster using an AWS Marketplace image, you must first subscribe through AWS. Subscribing to the offer provides you with the AMI id that the installation program uses to deploy worker nodes.
```
[the remark this answers is not in the document]
Comment (mjpytlak) [reply modules/installation-aws-marketplace-subscribe.adoc:20]: Done. Thank you.

File: modules/installation-creating-aws-worker.adoc [hunk 6]
```diff
@@ -114,10 +114,12 @@ endif::openshift-origin[]
 section of this topic and save it as a YAML file on your computer. This template
 describes the networking objects and load balancers that your cluster requires.
 
-. If you specified an `m5` instance type as the value for `WorkerInstanceType`,
+. Optional: If you specified an `m5` instance type as the value for `WorkerInstanceType`,
 add that instance type to the `WorkerInstanceType.AllowedValues` parameter
 in the CloudFormation template.
```
[the remark this answers is not in the document]
Comment (mjpytlak) [reply modules/installation-creating-aws-worker.adoc:119]: Done.

File: modules/installation-creating-aws-worker.adoc [hunk 7]
```diff
@@ -114,10 +114,12 @@ endif::openshift-origin[]
...
 
+. Optional: If you are deploying with an AWS Marketplace image, update the `Worker0.type.properties.ImageID` parameter with the AMI id you obtained from your subscription.
```
[the remark this answers is not in the document]
Comment (mjpytlak) [reply modules/installation-creating-aws-worker.adoc:121]: Done. Thank you.

File: modules/installation-aws-marketplace.adoc [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+Installing an {product-title} cluster using an AWS Marketplace image is available to customers who purchase the offer in North America.
+
+While the offer must be purchased in North America, you can deploy the cluster to any of the following supported paritions:
+
+* Public
+* GovCloud
+
+[NOTE]
+====
+Deploying a cluster using an AWS Marketplace image is not supported for the AWS secret regions or China regions.
```
[the remark this answers is not in the document]
Comment (mjpytlak) [reply modules/installation-aws-marketplace.adoc:18]: I am not sure if we have a hard and fast rule on this, as in many cases OpenShift Container Platform is implied is referenced elsewhere in modules, but I did add it.

File: modules/installation-aws-marketplace-subscribe.adoc (hunk 3 above)
Comment (mjpytlak) [reply modules/installation-aws-marketplace-subscribe.adoc:60]: Nice catch.

File: modules/installation-aws-marketplace-subscribe.adoc (hunk 4 above)
Comment (mjpytlak) [reply modules/installation-aws-marketplace-subscribe.adoc:33]: Not intended. Thank you. :)

**Comment by mjpytlak** [comment]
@bscott-rh You are good to back port this work into 4.9 and 4.8. Please coordinate with @jeana-redhat. Jeana will back port the respective work for machine sets when you are good to go.

**Pull request merged**